### ISSUE TITLE
Bug: В AppImage версии SS Classic VK не отображается Русский текст в отличие от Нативной и Flatpak версии

### ISSUE THREAD

**Issue by anzix** [tx00100xt/SeriousSamClassic-VK#47]
Система:

Operating System: Arch Linux
KDE Plasma Version: 6.2.4
KDE Frameworks Version: 6.8.0
Qt Version: 6.8.1
Kernel Version: 6.12.4-zen1-1-zen (64-bit)
Graphics Platform: Wayland
Processors: 12 × Intel® Xeon® CPU E5-1650 0 @ 3.20GHz
Memory: 15.5 ГиБ of RAM
Graphics Processor: AMD Radeon RX 580 Series

Проверено на:
SeriousSamTFE-Vk-1.10.6d-x86_64.AppImage
SeriousSamTSE-Vk-1.10.6d-x86_64.AppImage

Файлы игры из Steam версии, скопированы командой

```sh
cp -vfr "$HOME/.local/share/Steam/steamapps/common/Serious Sam Classic The First Encounter/"{*.gro,Levels,Help} ~/.local/share/Serious-Engine/serioussam
cp -vfr "$HOME/.local/share/Steam/steamapps/common/Serious Sam Classic The Second Encounter/"{*.gro,Levels,Help} ~/.local/share/Serious-Engine/serioussamse
```

Проблема: При запуске с файлами локализации текст отображается неправильной кодировкой и вообще не приминяется:

![Снимок экрана_20241217_191740](https://github.com/user-attachments/assets/6e1366be-f7e6-489d-bfbb-a44296bd7e83)
![Снимок экрана_20241217_191807](https://github.com/user-attachments/assets/059e638c-be4f-46f3-9a88-dd44ef11e6e7)

[Локализация которую я использовал](https://steamcommunity.com/sharedfiles/filedetails/?id=278147585) 

Gro файлы локализации я переместил по пути `~/.local/share/Serious-Engine/serioussam` и `~/.local/share/Serious-Engine/serioussamse`

Такое не происходит на Flatpak версии

**Comment by anzix** [comment]
Вот пример применённой локализации в Flatpak версии

![Снимок экрана_20241217_195538](https://github.com/user-attachments/assets/f6d55f98-3093-4cde-b950-b55450817305)

**Comment by tx00100xt** [comment]
Движок считывает гро архивы в алфавитном порядке c конца (Z-A). Если файлы из архива уже считаны, то другие с одинаковыми именами он пропускает. Нужно просто переименовать: 

TFE: "1_00c_Texts_RUS.gro" -> "Z_1_00c_Texts_RUS.gro"
TSE: "SE1_00_Texts_RUS.gro" -> "Z_SE1_00_Texts_RUS.gro"

**Comment by anzix** [comment]
Переименовываение не помогает (для оба TFE и TSE), текст всё также не применяется нормально. Вот `SeriousSam.log` лог загрузки гро файлов

```txt
...

Cannot load variable from 'DefaultMod.txt':
Cannot open file `/home/anix/.local/share/Serious-Engine/serioussam/DefaultMod.txt' (No such file or directory) ()
Current mod: <none>
Cannot load variable from 'ModExt.txt':
Cannot open file `/home/anix/.local/share/Serious-Engine/serioussam/ModExt.txt' (No such file or directory) ()
Loading group files...
  /tmp/.mount_SeriouAviRli/usr/share/serioussam/SE1_10.gro: 1817 files
  /home/anix/.local/share/Serious-Engine/serioussam/Z_1_00c_Texts_RUS.gro: 175 files
  /home/anix/.local/share/Serious-Engine/serioussam/1_04_patch.gro: 1 files
  /home/anix/.local/share/Serious-Engine/serioussam/1_00c_Sounds_RUS.gro: 60 files
  /home/anix/.local/share/Serious-Engine/serioussam/1_00c_scripts.gro: 28 files
  /home/anix/.local/share/Serious-Engine/serioussam/1_00c_Logo.gro: 1 files
  /home/anix/.local/share/Serious-Engine/serioussam/1_00c.gro: 2267 files
  /home/anix/.local/share/Serious-Engine/serioussam/1_00_music.gro: 55 files
  /home/anix/.local/share/Serious-Engine/serioussam/1_00_ExtraTools.gro: 58 files
...
```

**Comment by tx00100xt** [comment]
Да, потому что   SE1_10.gro загружается раньше чем Z_1_00c_Texts_RUS.gro.
Он внутри AppImage.

Можно попробовать распаковать Z_1_00c_Texts_RUS.gro. Насколько я помню гро загружаются после локальных файлов(не уверен).

**Comment by tx00100xt** [comment]
Получилось вот так:

![Image](https://github.com/user-attachments/assets/1c706744-89d0-4f3f-9b2c-bdd65c3421fd)

Потому что в нативной и флатпак версии используется 
https://github.com/tx00100xt/SeriousSamClassic-VK/blob/main/SamTFE/SE1_10b.gro

а в AppImage из Croteam repo: SE1_10.gro. Он более крупный и содержит битые текстуры и конфликтующие файлы с локализацией. Битые текстуры я исключил. Нужно еще конфликтующие файлы - дубликаты оттуда убрать.
SE1_10.gro вместо SE1_10b.gro используется для автономного запуска игры без самих файлов игры.

**Comment by anzix** [comment]
А, понятно, спасибо за объяснение. Если нужно тестирование текста, то можете написать здесь.
